macros/contextInequalities.pl

```diff
@@ -840,7 +840,7 @@ package Inequalities::MakeInterval;
 
 sub new {
   my $self = shift;
-  $self = $self->SUPER::new(@_);
+  $self = Value::Interval->new(@_);
```

Comment:
Would it be better to change this to
```suggestion
  $self = (shift)::Interval->new(@_);
```
and delete line 842 (as is done on line 856 for `Inequalities::List`)?  It seems that this makes the `$self` usage in this situation more clear.

Reply:
I think you meant `(shift)->SUPER::new(@_);`

Reply:
That is literally what is on line 856, but for this I meant `(shift)::Interval->new(@_)`.  I was referring to the `$self` usage and adapted that to this change.  But looking closely now, I see that both are wrong.  It looks like line 842 should just be deleted, and the `my` added to line 843.

Reply:
calling `Value::Interval->new` would be the same as `shift->SUPER::new` since this package @ISA `Value::Interval` so `my $self = (shift)->SUPER::new(@_);` would work the same as the two lines in the PR do now.

Reply:
No, the line is correct as is.  This is downgrading the `Inequalities::Interval` to a standard `Value::Interval`.  Neither of the suggestions would do that.  Also, removing the `my $self = shift` would leave the original object in the `@_` array, and that would be passed to `new()` method, which isn't correct.

Reply:
Sorry, my remark about shift is wrong.  D'oh.

Reply:
> calling `Value::Interval->new` would be the same as `shift->SUPER::new` since this package ISA `Value::Interval`

Just to be clear, that's not how it works.  The `new()` methods base their return values on the class of the caller, not on the `@ISA` value.  The `new()` methods typically start with

``` perl
  my $self = shift; my $class = ref($self) || $self;
``` 

and end with returning an object that is blessed using

``` perl
bless {...}, $class
```

So if you use `(shift)->SUPER::new(@_)`, that will produce an object with the same class as `(shift)` regardless of whether the `new()` is `SUPER::new()` or not.  Without that, every subclass would have to override `new()` and re-bless the return value to be its class (with it, subclasses are created as instances of the subclasses automatically).
It also allows `Value::Interval->new()` to produce an instance of `Value::Interval` since that is what `$class` will be in that case.  The `@ISA` value is used to method inheritance, not instance class creation.

Reply:
Aha, yes, I see. (I also see that I was effectively suggesting a return to the prior code...) This ensures that object created has the Value::Interval class rather than the Inequalities::MakeInterval class.

What I'm still unclear on regards the original forum post -- and I think belongs to the other code changes. I'll make a separate comment for that.